src/ast.ts

```diff
@@ -239,19 +246,57 @@ export const replaceVariableDefinitions = (
 };
 
 /**
- * return the path that will be returned in the response from from the chain of parents
+ * Return the path that will be returned in the response from the chain of parents.
+ * By default this will only build up paths for field nodes, but the anyKind flag allows
+ * to build paths for any named node.
+ *
+ * It also supports aliases.
  */
 /** @hidden */
-export const extractPath = (parents: ReadonlyArray<ASTNode>): ReadonlyArray<string> => {
+export const extractPath = (
+  parents: ReadonlyArray<ASTNode>,
+  anyKind?: boolean
```

Comment:
what's the reason for this `anyKind` param? What else besides a field would we want to get a path for?

Reply:
This is related to the matchAnyPath argument added to the Rewriter constructor. It's particularly thought for the Custom Rewriter, allowing to control other aspects of the query other than just Fields. 

This variable isn't very useful for the existing Rewriters since they are all meant for Field nodes.

Reply:
ah ok, that looks pretty cool!